### PR TITLE
libgit2: fix build issues

### PR DESCRIPTION
**Pull request by mmatongo**
## Existing package

- [Y] I am the maintainer of this package.

### CODE REVIEW

**Comment by mmatongo** [comment]
This should fix #300 @git-bruh

**Comment by git-bruh** [comment]
Hmm looks like i still need to pass `LDFLAGS` manually, the zlib option is [off](https://github.com/libgit2/libgit2/blob/main/CMakeLists.txt#L53) by default. Could you enable both shared and static libraries ?

**Comment by git-bruh** [comment]
When i try to build `libgit2` with `-static`, i get the errors:
```
ld: error: attempted static link of dynamic object /usr/lib/libssl.so
ld: error: attempted static link of dynamic object /usr/lib/libcrypto.so
ld: error: attempted static link of dynamic object /usr/lib/libz.so
```
which means its trying to link against shared libs for some reason, even though static libs are available for everything :thinking:

**Comment by mmatongo** [comment]
> When i try to build static `libgit2` libraries with `-static`, i get the errors:

I'm confused, why are you trying to build it with `-static` when shared libraries are disabled [here](https://github.com/mmatongo/repo-community/blob/ce10dd015a4bf29ceeb696385552b858becd0576/community/libgit2/build#L8) ?

**Comment by git-bruh** [comment]
Since i was getting undefined references, I thought maybe something is going wrong in the build so just tried a build with `-static` along with disabling shared libs. The build actually succeeds if you disable `BUILD_CLAR` but the undefined references still remain.

**Comment by mmatongo** [comment]
I get;
```sh
[100%] Linking C executable ../libgit2_clar
/usr/bin/ld: attempted static link of dynamic object `/usr/lib/libssl.so'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/libgit2_clar.dir/build.make:7481: libgit2_clar] Error 1
make[1]: *** [CMakeFiles/Makefile2:307: tests/CMakeFiles/libgit2_clar.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

When I try to build with `$LDFLAGS -static`

This is not needed seeing as shared libraries are off by default

**Comment by dilyn-corner** [comment]
Should probably tell `libgit2` to build with the static libraries for those dependencies - I imagine when you start the build `cmake` reports that the libcrypto it found is `/usr/lib/libcryptos.so`, for instance.

The thing about `libgit` is its going to be an absolute *unit* of a lib if you build it statically (I mean, 2MB lib.a, but still). `stagit` devs [recommend disabling a ton of stuff](https://git.codemadness.org/stagit/file/README.html#l51) if you build it statically, for instance.

The solution here is to either add:
```
    -DZLIB_LIBRARY=/usr/lib/libz.a \
    -DOPENSSL_SSL_LIBRARY=/usr/lib/libssl.a \
    -DOPENSSL_CRYPTO_LIBRARY=/usr/lib/libcrypto.a \
```
To the build file, or simply build the shared library - except this doesn't resolve my issues building `stagit` with a static `libgit2` for instance:
`
/usr/bin/ld: /usr/lib/gcc/x86_64-pc-linux-musl/10.2.0/../../../../lib/libgit2.a(crypt_openssl.c.o): in function `ntlm_hmac_ctx_free':
(.text+0x1d1): undefined reference to `HMAC_CTX_free'
`

I haven't yet bothered to dive further into this but I could if you'd like. `stagit` builds if I specify `-lz -lcrypto -lssl` in the `LDFLAGS` during `make`, though (as expected - we're telling `stagit` to not rely on `libgit2.a` for these libs, because for some reason the symbols aren't making it in).

BTW: the `-static` (C)flag just tells the linker to link against static libraries instead of shared ones. So git-bruh attempted the right thing.

**Comment by mmatongo** [comment]
I guess I'll just bundle both shared and static libs into the package.

**Comment by mmatongo** [comment]
New manifest
```sh
/var/db/kiss/installed/libgit2/version
/var/db/kiss/installed/libgit2/sources
/var/db/kiss/installed/libgit2/manifest
/var/db/kiss/installed/libgit2/depends
/var/db/kiss/installed/libgit2/checksums
/var/db/kiss/installed/libgit2/build
/var/db/kiss/installed/libgit2/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/lib/pkgconfig/libgit2.pc
/usr/lib/pkgconfig/
/usr/lib/libgit2.so.1.1.0
/usr/lib/libgit2.so.1.1
/usr/lib/libgit2.so
/usr/lib/libgit2.a
/usr/lib/
/usr/include/git2/worktree.h
/usr/include/git2/version.h
/usr/include/git2/types.h
/usr/include/git2/tree.h
/usr/include/git2/transport.h
/usr/include/git2/transaction.h
/usr/include/git2/trace.h
/usr/include/git2/tag.h
/usr/include/git2/sys/transport.h
/usr/include/git2/sys/stream.h
/usr/include/git2/sys/repository.h
/usr/include/git2/sys/refs.h
/usr/include/git2/sys/reflog.h
/usr/include/git2/sys/refdb_backend.h
/usr/include/git2/sys/path.h
/usr/include/git2/sys/openssl.h
/usr/include/git2/sys/odb_backend.h
/usr/include/git2/sys/merge.h
/usr/include/git2/sys/mempack.h
/usr/include/git2/sys/index.h
/usr/include/git2/sys/hashsig.h
/usr/include/git2/sys/filter.h
/usr/include/git2/sys/diff.h
/usr/include/git2/sys/credential.h
/usr/include/git2/sys/cred.h
/usr/include/git2/sys/config.h
/usr/include/git2/sys/commit.h
/usr/include/git2/sys/alloc.h
/usr/include/git2/sys/
/usr/include/git2/submodule.h
/usr/include/git2/strarray.h
/usr/include/git2/stdint.h
/usr/include/git2/status.h
/usr/include/git2/stash.h
/usr/include/git2/signature.h
/usr/include/git2/revwalk.h
/usr/include/git2/revparse.h
/usr/include/git2/revert.h
/usr/include/git2/reset.h
/usr/include/git2/repository.h
/usr/include/git2/remote.h
/usr/include/git2/refspec.h
/usr/include/git2/refs.h
/usr/include/git2/reflog.h
/usr/include/git2/refdb.h
/usr/include/git2/rebase.h
/usr/include/git2/proxy.h
/usr/include/git2/pathspec.h
/usr/include/git2/patch.h
/usr/include/git2/pack.h
/usr/include/git2/oidarray.h
/usr/include/git2/oid.h
/usr/include/git2/odb_backend.h
/usr/include/git2/odb.h
/usr/include/git2/object.h
/usr/include/git2/notes.h
/usr/include/git2/net.h
/usr/include/git2/message.h
/usr/include/git2/merge.h
/usr/include/git2/mailmap.h
/usr/include/git2/indexer.h
/usr/include/git2/index.h
/usr/include/git2/ignore.h
/usr/include/git2/graph.h
/usr/include/git2/global.h
/usr/include/git2/filter.h
/usr/include/git2/errors.h
/usr/include/git2/diff.h
/usr/include/git2/describe.h
/usr/include/git2/deprecated.h
/usr/include/git2/credential_helpers.h
/usr/include/git2/credential.h
/usr/include/git2/cred_helpers.h
/usr/include/git2/config.h
/usr/include/git2/common.h
/usr/include/git2/commit.h
/usr/include/git2/clone.h
/usr/include/git2/cherrypick.h
/usr/include/git2/checkout.h
/usr/include/git2/cert.h
/usr/include/git2/buffer.h
/usr/include/git2/branch.h
/usr/include/git2/blob.h
/usr/include/git2/blame.h
/usr/include/git2/attr.h
/usr/include/git2/apply.h
/usr/include/git2/annotated_commit.h
/usr/include/git2/
/usr/include/git2.h
/usr/include/
/usr/
```

Should be a fine @git-bruh based on what @dilyn-corner has put forth.

File: community/libgit2/build [hunk 1]
```diff
@@ -2,10 +2,21 @@
 
 export DESTDIR="$1"
 
-cmake -B build \
+cmake -B build-shared \
```
Comment (dilyn-corner) [inline community/libgit2/build:5]: should be build-static because for this one we're not building shared libs

File: community/libgit2/build [hunk 2]
```diff
@@ -2,10 +2,21 @@
...
+    -DBUILD_SHARED_LIBS=OFF \
+    -DUSE_BUNDLED_ZLIB=OFF
 
-cmake --build build
-cmake --install build
+cmake --build build-shared
+cmake --install build-shared
+
+# build shared libs https://github.com/kiss-community/repo-community/pull/304
+cmake -B build-static \
```
Comment (dilyn-corner) [inline community/libgit2/build:15]: Should be build-shared because ^^^

Comment (mmatongo) [reply community/libgit2/build]: fixed

File: community/libgit2/build (hunk 1 above)
Comment (mmatongo) [reply community/libgit2/build]: fixed

**Pull request merged**